src/extension/extensionServer.ts

```diff
@@ -123,6 +124,19 @@ export class ExtensionServer implements vscode.Disposable {
         return Q.resolve<void>(void 0);
     }
 
+    /**
+     * Message handler for OPEN_FILE_AT_LOCATION
+     */
+    private openFileAtLocation(filename: string, lineNumber: number): Q.Promise<any> {
+        return Q(vscode.workspace.openTextDocument(vscode.Uri.file(filename)).then(document => {
+            return vscode.window.showTextDocument(document).then(editor => {
```

Comment:
`vscode.workspace.openTextDocument` will return a `thenable`. This type should be compatible with our Q promises, therefore Q should be able to convert it.